src/Parsers/LexerStandalone.h

```diff
@@ -0,0 +1,185 @@
+#pragma once
+
+/// This allows building Lexer without any dependencies or includes for WebAssembly or Emscripten.
```

Comment:
isn't too much of code duplication is going on here? maybe we can unify it somehow...